### ISSUE TITLE
RequireEmptyLineBeforeBlockTagGroup: false positive on line with only single space

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#9685]
````
$ cat TestClass.java
public class TestClass {
    /**
     * Method.
     * 
     * @throws Exception
     *             if there is an error.
     */
    void method() throws Exception {
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="RequireEmptyLineBeforeBlockTagGroup"/>
    </module>
</module>

$ java -jar checkstyle-8.41-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:5: Javadoc tag '@throws' should be preceded with an empty line. [RequireEmptyLineBeforeBlockTagGroup]
Audit done.
Checkstyle ends with 1 errors.
````

It is not clear from the report, but the line 5 contains only spaces and the asterisk. What makes this line special is that there is an extra space after the asterisk on the line.

Eclipse's javadoc creation likes to put in these spaces when using it to start the Javadoc.

I am looking for lines like this to not be reported. The line is empty only containing the asterisk and spaces.

**Comment by pbludov** [comment]
I'm agree. Invisible trailing whitespace should be handled by another check. From the perspective of this check, any line of non-printable symbols is a blank line.

**Comment by AkMo3** [comment]
@rnveach @pbludov This might be fixed in #10042 as the report shows, should this be linked to that PR?

**Comment by pbludov** [comment]
@AkMo3 please take a look.

The AST is
```
        |   |   |   |--COMMENT_CONTENT -> *\n     * Method.\n     * \n     * @throws Exception\n     *             if there is an error.\n      [3:6]
        |   |   |   |   `--JAVADOC -> JAVADOC [3:7]
        |   |   |   |       |--NEWLINE -> \n [3:7]
        |   |   |   |       |--LEADING_ASTERISK ->      * [4:0]
        |   |   |   |       |--TEXT ->  Method. [4:6]
        |   |   |   |       |--NEWLINE -> \n [4:14]
        |   |   |   |       |--LEADING_ASTERISK ->      * [5:0]
        |   |   |   |       |--TEXT ->   [5:6]
        |   |   |   |       |--NEWLINE -> \n [5:7]
        |   |   |   |       |--LEADING_ASTERISK ->      * [6:0]
        |   |   |   |       |--WS ->   [6:6]
        |   |   |   |       |--JAVADOC_TAG -> JAVADOC_TAG [6:7]
        |   |   |   |       |   |--THROWS_LITERAL -> @throws [6:7]
```
Lines 4 and 5 should have the `WS` tag after `LEADING_ASTERISK`. Line 6 is good.

**Comment by liudas-stonys** [comment]
After 3 years... Issue still persists. Good work, guys! :D

**Comment by romani** [comment]
@liudas-stonys, please contribute a fix, we are welcoming all to participate. We have almost 800 issues, please help us to make it less by one ;)

**Comment by romani** [comment]
We can make violation message to show what was found instead of empty line.
`Javadoc tag '@throws' should be preceded with an empty line, found '   '`.

It can be new violation message, if we think it makes sense only for invisible symbols.

**Comment by rnveach** [comment]
@romani 

> We can make violation message to show 

That will not work for me. As I reported, Eclipse likes to add these spaces to the Javadoc by default and I didn't find an option to turn them off.

I am looking for whitespaces to be treated as an empty line and no violation. If someone wants to remove trailing spaces, they use RegExp like we do in this project.

**Comment by romani** [comment]
In this case it is issue on Eclipse side, or missed config in Eclipse to remove trailing spaces.

Proposal for message extension at least can help users to spot a problem. Our project is so strict on all aspects of code, I am not sure we need to relax it here.
